remix-analyzer/src/solidity-analyzer/modules/gasCosts.ts

```diff
@@ -1,72 +1,96 @@
-        if (gas === null || parseInt(gas) >= 3000000 || gas === 'infinite') {
-          report.push({
-            warning: `Gas requirement of function ${contract.name}.${functionName} ${gasString}. 
-            If the gas requirement of a function is higher than the block gas limit, it cannot be executed.
-            Please avoid loops in your functions or actions that modify large areas of storage
-            (this includes clearing or copying arrays in storage)`
-          })
+    })
+    for (const method of methodsWithSignature) {
+      for (const filename in compilationResults.contracts) {
+        for (const contractName in compilationResults.contracts[filename]) {
+          const contract: CompiledContract = compilationResults.contracts[filename][contractName]
+          const methodGas: Record<string, any> | undefined = this.checkMethodGas(contract, method.signature)
+          if(methodGas && methodGas.isInfinite) {
+            if(methodGas.isFallback) {
+              report.push({
+                warning: `Fallback function of contract ${contractName} requires too much gas (${methodGas.msg}). 
+                If the fallback function requires more than 2300 gas, the contract cannot receive Ether.`,
+                location: method.src
+              })
+            } else {
+              report.push({
+                warning: `Gas requirement of function ${contractName}.${method.name} ${methodGas.msg}. 
+                If the gas requirement of a function is higher than the block gas limit, it cannot be executed.
+                Please avoid loops in your functions or actions that modify large areas of storage
+                (this includes clearing or copying arrays in storage)`,
+                location: method.src
+              })
+            } 
+          } else continue
         }
       }
-    })
+    }
     return report
   }
 
-  /**
-  * call the given @arg cb (function) for all the contracts. Uses last compilation result
-  * stop visiting when cb return true
-  * @param {Function} cb    - callback
-  */
-  // @TODO has been copied from remix-ide repo ! should fix that soon !
-  private visitContracts (contracts: CompiledContractObj | undefined, cb: ((contract: VisitedContract) => void | undefined)): void {
-    for (let file in contracts) {
-      for (let name in contracts[file]) {
-        if (cb({ name: name, object: contracts[file][name], file: file })) return
+  // To create the method signature similar to contract.evm.gasEstimates.external object
+  // For address payable, return address 
+  private getSplittedTypeDesc(node: VariableDeclarationAstNode): string {
+    return node.typeDescriptions.typeString.split(' ')[0]
+  }
+
+
+  private checkMethodGas(contract: CompiledContract, methodSignature: string): Record<string, any> | undefined {
+    if(contract.evm && contract.evm.gasEstimates && contract.evm.gasEstimates.external) {
```

Comment:
it seems to not return all the cases.. at least it only returns `isInfinite: true,`  which is strange

Reply:
Yes, it only checks if method gas is infinite. In other cases, we just continue to move further, see: https://github.com/ethereum/remix/pull/1487/files#diff-863462ce97d02e9d05a6a6f825c8ace8R58

Reply:
ah ok so it is reporting only when the function has infinite gas?

Reply:
yes, may be function name can be updated.